packages/x-data-grid/src/material/variables.ts

```diff
@@ -23,58 +23,59 @@ function transformTheme(t: Theme): GridCSSVariablesInterface {
-
-    [k.colors.interactive.hover]: removeOpacity(t.palette.action.hover),
-    [k.colors.interactive.hoverOpacity]: t.palette.action.hoverOpacity,
-    [k.colors.interactive.focus]: removeOpacity(t.palette.primary.main),
-    [k.colors.interactive.focusOpacity]: t.palette.action.focusOpacity,
-    [k.colors.interactive.disabled]: removeOpacity(t.palette.action.disabled),
-    [k.colors.interactive.disabledOpacity]: t.palette.action.disabledOpacity,
+    [k.colors.foreground.base]: (t.vars || t).palette.text.primary,
+    [k.colors.foreground.muted]: (t.vars || t).palette.text.secondary,
+    [k.colors.foreground.accent]: (t.vars || t).palette.primary.dark,
+    [k.colors.foreground.disabled]: (t.vars || t).palette.text.disabled,
+
+    [k.colors.interactive.hover]: removeOpacity((t.vars || t).palette.action.hover),
+    [k.colors.interactive.hoverOpacity]: (t.vars || t).palette.action.hoverOpacity,
+    [k.colors.interactive.focus]: removeOpacity((t.vars || t).palette.primary.main),
+    [k.colors.interactive.focusOpacity]: (t.vars || t).palette.action.focusOpacity,
+    [k.colors.interactive.disabled]: removeOpacity((t.vars || t).palette.action.disabled),
+    [k.colors.interactive.disabledOpacity]: (t.vars || t).palette.action.disabledOpacity,
     [k.colors.interactive.selected]: selectedColor,
-    [k.colors.interactive.selectedOpacity]: t.palette.action.selectedOpacity,
+    [k.colors.interactive.selectedOpacity]: (t.vars || t).palette.action.selectedOpacity,
 
     [k.header.background.base]: backgroundHeader,
     [k.cell.background.pinned]: backgroundPinned,
 
-    [k.radius.base]:
-      typeof t.shape.borderRadius === 'number' ? `${t.shape.borderRadius}px` : t.shape.borderRadius,
+    [k.radius.base]: radius,
 
     [k.typography.fontFamily.base]: t.typography.fontFamily as string,
     [k.typography.fontWeight.light]: t.typography.fontWeightLight as string,
     [k.typography.fontWeight.regular]: t.typography.fontWeightRegular as string,
     [k.typography.fontWeight.medium]: t.typography.fontWeightMedium as string,
     [k.typography.fontWeight.bold]: t.typography.fontWeightBold as string,
-    [k.typography.body.fontFamily]: t.typography.body2.fontFamily as string,
-    [k.typography.body.fontSize]: t.typography.body2.fontSize as string,
-    [k.typography.body.fontWeight]: t.typography.body2.fontWeight as string,
-    [k.typography.body.letterSpacing]: t.typography.body2.letterSpacing as string,
-    [k.typography.body.lineHeight]: t.typography.body2.lineHeight as string,
-    [k.typography.small.fontFamily]: t.typography.caption.fontFamily as string,
-    [k.typography.small.fontSize]: t.typography.caption.fontSize as string,
-    [k.typography.small.fontWeight]: t.typography.caption.fontWeight as string,
-    [k.typography.small.letterSpacing]: t.typography.caption.letterSpacing as string,
-    [k.typography.small.lineHeight]: t.typography.caption.lineHeight as string,
```

Comment:
We have to condense these to a single `font` property to support the material UI theme as they do not expose individual font properties like this as CSS variables.